lib/patch_status_json.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+
+"""
+Patch status.json file to avoid tracebacks for non-root from `ua status`
+
+Between 27.1 and 27.2 status.json broke schema key backward-compatibility.
+
+Running `ua status` as non-root users does not generate a new status.json with
+the updateed status.json schema so the logic which renders ua status
+information will traceback on KeyErrors.
+
+Since we haven't introduced official schema versioning in UA client yet,
+this patch is a stop-gap until official schema version handling is
+delivered.
+"""
+
+import copy
+import json
+import logging
+
+from uaclient.cli import setup_logging
+from uaclient import util
+
+
+def patch_status_json_schema_0_1(status_file: str):
+    """Patch incompatible status.json file schema to align with version 0.1."""
+    setup_logging(logging.INFO, logging.DEBUG)
+    content = util.load_file(status_file)
+    try:
+        status = json.loads(content, cls=util.DatetimeAwareJSONDecoder)
+    except ValueError as e:
+        logging.debug(
+            "Unable to patch /var/lib/ubuntu-advantage/status.json: {}".format(
+                str(e)
+            )
+        )
+        return
+    new_status = copy.deepcopy(status)
+    if float(status.get("_schema_version", 0)) >= 0.1:
+        return  # Already have version 0.1 from daily PPA
+    logging.debug("Patching /var/lib/ubuntu-advantage/status.json schema")
+    new_status["_schema_version"] = "0.1"
+    new_status["account"] = {
```

Comment:
I don't know if we want to do that, but maybe we should consider adding the new keys too, such as `created_at` in `account`  and `contract`

Reply:
I was trying to hold off on the full schema key upgrades here and only perform the minimal key updates needed to allow non-root to succeed with status calls. I wanted to explicitly avoid performing a full schema update here for anything that is unneeded for status. The reason being that any further ua commands run by root will be correcting and rewriting this file anyway with the proper full schema.